common/program_synthesis/dsl.py

```diff
@@ -47,6 +47,10 @@ def toggle_constraints(self):
 
         self._constraints_enabled = not self._constraints_enabled
 
+    def to_string(self):
```

Comment:
use magic \_\_repr\_\_ method instead

Reply:
why __repr__ and not __str__?

Reply:
I guess we can do both: "The str() function returns a user-friendly description of an object. The repr() method returns a developer-friendly string representation of an object." Since it's pretty user friendly I'll do __str__ indeed

Reply:
Alright, agreed

Reply:
Let's do that in the next PR